### PR TITLE
[BACKLOG-28010] Add new file open/save dialog

### PR DESCRIPTION
**Pull request by bmorrise**

### CODE REVIEW

**Comment by bmorrise** [comment]
@ddiroma

File: plugins/file-open-save-new/api/src/main/java/org/pentaho/di/plugins/fileopensave/api/providers/Utils.java [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+import java.util.List;
+
+/**
+ * Created by bmorrise on 2/23/19.
+ */
+public class Utils {
+
+  public static List<String>
+    validExtensions = Arrays
+    .asList( "dsn", "ph", "config", "aze", "rom", "bowerrc", "ktr", "ini", "kts", "p7b", "docm", "rtf", "rub", "ico",
```
Comment (ddiroma) [inline plugins/file-open-save-new/api/src/main/java/org/pentaho/di/plugins/fileopensave/api/providers/Utils.java]: Please alphabetize these

File: plugins/file-open-save-new/core/src/main/java/org/pentaho/di/plugins/fileopensave/dialog/FileOpenSaveDialog.java [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+    parentPath = (String) arguments[ 3 ];
+    connection = (String) arguments[ 4 ];
+    provider = (String) arguments[ 5 ];
+    type = (String) arguments[ 6 ];
+  }
+
+  private static String getClientPath() {
+    Properties properties = new Properties();
+    try {
+      InputStream inputStream =
```
Comment (ddiroma) [inline plugins/file-open-save-new/core/src/main/java/org/pentaho/di/plugins/fileopensave/dialog/FileOpenSaveDialog.java]: does this stream need to be closed?

File: plugins/file-open-save-new/api/src/main/java/org/pentaho/di/plugins/fileopensave/api/providers/Utils.java (hunk 1 above)
Comment (bmorrise) [reply plugins/file-open-save-new/api/src/main/java/org/pentaho/di/plugins/fileopensave/api/providers/Utils.java]: Done.

File: plugins/file-open-save-new/core/src/main/java/org/pentaho/di/plugins/fileopensave/dialog/FileOpenSaveDialog.java (hunk 2 above)
Comment (bmorrise) [reply plugins/file-open-save-new/core/src/main/java/org/pentaho/di/plugins/fileopensave/dialog/FileOpenSaveDialog.java]: Yes, it should

**Pull request merged**